### PR TITLE
[besu] update flux delete role

### LINKED ISSUE #2157: bug(shared): intermittent flux installation/uninstallation issues
Description:
As a developer, I want to fix the flux resources going in terminating state forever.

Steps to reproduce the behaviour:
1. Reset and redeploy the network on which you are working like besu, fabric, etc.
2. It will throw an error at the time of terminating the namespace in which flux is installed.
3. Check the status of the namespace using the command `kubectl get ns <name of namespace>`
4. You will find your namespace as `Terminating`

Expected behaviour:
Namespace should be deleted and the reset-network should happen successfully.

Screenshot:
Image-FluxTerminating

**Acceptance Criterion :**

1. Reset working as expected for platforms

- [x] HL Besu
- [x] GoQuorum
- [x] R3 Corda OS
- [x] R3 Corda Enterprise
- [x] HL Indy
- [x] HL Fabric

### PR DESCRIPTION
**Pull request by saurabhkumarkardam**
**bug(besu): updated reset.yaml** 

Added patching and deletion code for the following kubernetes resources
 1. ClusterRole 
 2. ClusterRoleBinding 
 3. GitRepository 
 4. Kustomization 
 5. HelmChart 
 6. HelmRelease
 7. Organizations namespaces
 8. flux namespace
 
fixes #2157

### CODE REVIEW

File: platforms/hyperledger-besu/configuration/reset-network.yaml [hunk 1]
```diff
@@ -110,7 +110,72 @@
         - "{{ network['organizations'] }}"
       ignore_errors: yes
 
-    ##Delete Flux namespace 
+    # Delete ClusterRoleList related to flux
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-besu/configuration/reset-network.yaml:113]: Please change it to `Delete clusterrole ...`

File: platforms/hyperledger-besu/configuration/reset-network.yaml [hunk 2]
```diff
@@ -110,7 +110,72 @@
...
+        kubeconfig: "{{ kubernetes.config_file }}"
+      vars:
+        kubernetes: "{{ item[1].k8s }}"
+      with_nested:
+        - ["crd-controller-flux-{{ network.env.type }}"]
+        - "{{ network['organizations'] }}"
+      ignore_errors: yes
+
+    # Delete ClusterRoleBinding related to flux
+    - name: Delete flux ClusterRoleBindingList
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-besu/configuration/reset-network.yaml:130]: Please change it to `ClusterRoleBinding`

File: platforms/hyperledger-besu/configuration/reset-network.yaml [hunk 3]
```diff
@@ -110,7 +110,72 @@
...
+      k8s:
+        state: absent
+        api_version: v1
+        kind: ClusterRoleBindingList
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-besu/configuration/reset-network.yaml:134]: Should it not be `ClusterRoleBinding` ?

File: platforms/hyperledger-besu/configuration/roles/delete/helm_releases/tasks/nested.yaml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# Storing release name and its namespace in an array
+- name: Nested loop for helm releases
+  set_fact:
+    helm_result={{ helm_result | default([]) + [{'release_name':helm_resource.spec.releaseName, 'ns':helm_resource.metadata.namespace}] }}
+  loop: "{{ resources.resources | list }}"
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-besu/configuration/roles/delete/helm_releases/tasks/nested.yaml:5]: I don't think this is tested.
There is no variable named `resources` here.
Also the variable `helm_releases_raw` created in main.yaml doesnt seem to be used here.

Please check and verify this.

**Review by jagpreetsinghsasan — CHANGES_REQUESTED** [review]

File: platforms/hyperledger-besu/configuration/roles/delete/helm_releases/tasks/nested.yaml (hunk 4 above)
Comment (saurabhkumarkardam) [reply platforms/hyperledger-besu/configuration/roles/delete/helm_releases/tasks/nested.yaml:5]: Hi @jagpreetsinghsasan 

Yes, that part of the code was missing. Thank you for letting me know about that. Now, the code has been updated and tested successfully.

File: platforms/hyperledger-besu/configuration/reset-network.yaml (hunk 1 above)
Comment (saurabhkumarkardam) [reply platforms/hyperledger-besu/configuration/reset-network.yaml]: Hi @jagpreetsinghsasan 

Updated to "# Delete ClusterRole related to flux" in the latest Pull Request. Thank you.

File: platforms/hyperledger-besu/configuration/reset-network.yaml (hunk 2 above)
Comment (saurabhkumarkardam) [reply platforms/hyperledger-besu/configuration/reset-network.yaml]: Hi @jagpreetsinghsasan 

Update to "ClusterRoleBinding" in the latest Pull Request. Thank you.

File: platforms/hyperledger-besu/configuration/reset-network.yaml (hunk 3 above)
Comment (saurabhkumarkardam) [reply platforms/hyperledger-besu/configuration/reset-network.yaml]: Hi @jagpreetsinghsasan

Update to "ClusterRoleBinding" in the latest Pull Request. Thank you.

File: platforms/hyperledger-besu/configuration/reset-network.yaml [hunk 5]
```diff
@@ -121,7 +162,8 @@
       vars:
         kubernetes: "{{ item.k8s }}"
       loop: "{{ network['organizations'] }}"
-      ignore_errors: yes
+      when: patch_result is succeeded
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-besu/configuration/reset-network.yaml:165]: Why do we have this `when` condition?
(Its fine to put the condition considering the other case `when: !patch_result is succeeded` is also tested or covered)

**Review by jagpreetsinghsasan — CHANGES_REQUESTED** [review]

**Review by suvajit-sarkar — APPROVED** [review]
LGTM

**Review by adityajoshi12 — APPROVED** [review]

**Review by jagpreetsinghsasan — APPROVED** [review]
LGTM

**Pull request merged**